ros/src/twist_controller/twist_controller.py

```diff
@@ -17,53 +17,70 @@ def __init__(self, steer_ratio, decel_limit, accel_limit, max_steer_angle, wheel
         self.max_let_accel = max_let_accel
         self.wheel_base = wheel_base
 
+        self.throttle_pid = PID(.15, .00, 0, self.decel_limit, self.accel_limit)
+        self.brake_pid = PID(15, 1, 0, 0, 100)
         self.steer_pid = PID(2.0, 0.0005, 2.0, -self.max_steer_angle, self.max_steer_angle)
-        self.throttle_pid = PID(1, 0.1, 0.1, self.decel_limit, self.accel_limit)
-        self.brake_pid = PID(1, 0.1, 0.1, self.decel_limit, self.accel_limit)
         self.last_velocity_error = 0
         self.last_time = 0
         self.DT = DT
+        self.brakeLatch = False
 
         self.yaw_controller = YawController(self.wheel_base, self.steer_ratio,
                                             0, self.max_let_accel, self.max_steer_angle)
 
     def control(self, target_linear_velocity, target_angular_velocity,
-                current_linear_velocity, dbw_status):
+                current_linear_velocity, dbw_status, log_handle):
         '''Defines target throttle, brake and steering values'''
 
         if dbw_status:
-
             # Update DT
             new_time = rospy.get_rostime()
             if self.last_time:  # The first time, we are not able to calculate DT
                 self.DT = (new_time - self.last_time).to_sec()
             self.last_time = new_time
 
             velocity_error = target_linear_velocity - current_linear_velocity
-
-            if self.is_change_acc(velocity_error):
-                self.throttle_pid.reset()
-                self.brake_pid.reset()
-
-            # implement throttle controller
-            if velocity_error >= 0:
-                throttle = self.throttle_pid.step(velocity_error, DT)
+            pid_throttle, feedforward_throttle, decel_target = 0, 0, 0
+
+            # implement longitudinal controller
+
+            # if we're going too slow, release the brakes (if they are applied)
+            #T his essentially adds hysterisis: the brakes are only enabled if our velocity error is negative, and only
+            # released if the velocity error is positive 2 or greater.
+            if velocity_error > 2:
+                self.brakeLatch = False
+            if self.brakeLatch is False:
+                pid_throttle = self.throttle_pid.step(velocity_error, DT, log_handle)
+                feedforward_throttle = target_linear_velocity*.01888 #based on desired speed, predict how much throttle we need at steady state
+                throttle = pid_throttle + feedforward_throttle
+                accel_limit = 1 #mps2
+                maxThrottle = 0.1962*accel_limit+0.083 # max throttle allowed for a given acceleration limit
```

Comment:
I wonder if those hard coded values are only based on the simulation or will also be valid on Carla